Duplicati.BackupExplorer/ViewModels/MainViewModel.cs

```
@@ -397,6 +437,112 @@ public async Task SelectDatabase(object parent)
+                fileSize = blocks.Sum(x => x.Size);
+            }
+
+            ft.AddPath(Path.Join(file.Prefix, file.Path), file.BlocksetId, fileSize);
+            Progress += detailedProgress;
+        }
+
+        var node = _loadedTrees.Find(backup.Fileset.Id);
+        if (node != null)
+        {
+            // Move to front of queue
+            _loadedTrees.Remove(node);
+            _loadedTrees.AddFirst(node);
+        }
+        else
+        {
+            _loadedTrees.AddFirst(backup.Fileset.Id);
+            if (_loadedTrees.Count > MAX_LOADED_FILESETS)
+            {
+                // Unload the latest fileset
+                var id = _loadedTrees.Last!.Value;
+                var lastBackup = Backups.FirstOrDefault(b => b.Fileset.Id == id);
+                if (lastBackup != null)
+                {
+                    lastBackup.FileTree = null;
+                    _loadedTrees.RemoveLast();
+                }
+            }
+        }
+        backup.FileTree = ft;
+
+    }
+    void LoadFilesetStats(Backup backup)
+    {
+        ProgressTextFormat = $"Loading fileset stats {backup} ({{1:0}} %)";
+        // Load only the overview statistics for the backup
+        if (backup.FileTree != null)
+        {
+            // Already loaded
+            return;
+        }
+        backup.Size = _database.GetFilesetSize(backup.Fileset.Id);
+    }
+
+    // Runs in background thread and loads the filetree for _loadingFileTree
+    private async Task LazyLoadFileTrees(CancellationToken token)
+    {
+        try
+        {
+            while (true)
+            {
+                if (!_loadingWait.WaitOne(100) || IsLoadingDatabase)
+                {
+                    token.ThrowIfCancellationRequested();
+                    continue;
+                }
+                var backup = _loadingFileTree;
+                _loadingFileTree = null;
+                _loadingWait.Reset();
+                if (backup == null)
+                {
```

When would `backup` be `null` here? I think it can only happen in case of the mentioned race conditions? I think it is an error when it occurs and should not be silently ignored.